### PR TITLE
build(ci): automate release version

### PR DESCRIPTION
**Pull request by inc-man**
# Motivation

We want an automated workflow for release and tag management

# Changes

- created version bump workflow
- created tag / release workflow

# Tests

none so far

### CODE REVIEW

**Review by malaaf81 — COMMENTED** [review]
Hint: use "-" or "_" as delimiter for filename. Both looks ugward
The workflows itself look good/correct for PR-based versioning

File: .github/workflows/version_bump.yml
Comment (AntonioVentilii-DFINITY) [inline .github/workflows/version_bump.yml:33]: for this step and the above, better to use `prepare` action

**Review by AntonioVentilii-DFINITY — COMMENTED** [review]
A few comments for the code per se, nothing particular.

But for release notes / changelog and how to do that I will defer to @bitdivine and @peterpeterparker , since they have more experience with DFINITY repos and releases

So, don't wait for my approval when they will review

File: .github/workflows/tag-release.yml
Comment (inc-man) [reply .github/workflows/tag-release.yml:5]: it works together with the if statement. from my pov release branches should always be merged.

File: .github/workflows/version_bump.yml
Comment (inc-man) [reply .github/workflows/version_bump.yml:1]: done

File: .github/workflows/tag-release.yml
Comment (inc-man) [reply .github/workflows/tag-release.yml]: done

File: .github/workflows/version_bump.yml
Comment (inc-man) [reply .github/workflows/version_bump.yml]: done

File: .github/workflows/tag-release.yml
Comment (AntonioVentilii-DFINITY) [reply .github/workflows/tag-release.yml:5]: the if statement does not filter for merged/unmerged; and we cannot suppose that all bumping PRs will be merged. There may be the case that we created it, and then we close it because maybe it is still not time, or whatever other reason, a mistake, etc

Comment (AntonioVentilii-DFINITY) [reply .github/workflows/tag-release.yml:5]: sorry, my bad, I read the IF statement wrong, you filter for it! my bad

Comment (AntonioVentilii-DFINITY) [reply .github/workflows/tag-release.yml:5]: but so, just to confirm, ONLY the PRs that are merged will be treated with this workflow, right?

Comment (inc-man) [reply .github/workflows/tag-release.yml:5]: yes exactly the workflow will run on closed prs / if merged and release in its name.

**Review by bitdivine — APPROVED** [review]

**Pull request merged**
